pkg/create/templates/scripts.go

```diff
@@ -20,11 +20,11 @@ fi
 #
 if [ "$(ls /tmp/artifacts/ 2>/dev/null)" ]; then
   echo "---> Restoring build artifacts..."
-  mv /tmp/artifacts/. ./
+  mv /tmp/artifacts/* ./
 fi
 
 echo "---> Installing application source..."
-cp -Rf /tmp/src/. ./
+cp -Rf /tmp/src/* ./
```

Comment:
I mean any hidden folder that is part of the source repository being built/operated on by the assemble script.  The existing logic will include those dirs, your logic will not.

s2i create is just a helper to give you some skeleton/example files when you are starting development of a new s2i builder image.  As you say, it's not a critical feature if you already know how to write the scripts.

As for why s2i create creates the .s2i folder as a hidden folder, probably not for any good reason, it could well be created as non-hidden instead and probably should be.